static/usr/lib/systemd/system/rtc-sys-time-init.service

```diff
@@ -0,0 +1,29 @@
+# rtc-sys-time-init.service is enabled by rtc-sys-time-init-generator.
+# It is disabled by default, to enable it boot with kernel command line 
+# 'rtc_sys_time_init'. When enabled the generator hooks 
+# rtc-sys-time-init.service into systemd-timesyncd.service with symlink 
+# in /run/systemd/generator*/systemd-timesyncd.service.wants.
+# 
+# This service address two different scenarios in order:
+#  a.) Ensure system time is not behind the most recent epoch/clock file timestamps.
+#      This addresses: https://github.com/snapcore/core20/pull/136
+#  b.) Set system time to valid RTC time when kernel configuration hctosys is enabled
+#      with kernel configuration CONFIG_RTC_HCTOSYS and hctosys device is specified 
+#      with kernel configuration CONFIG_RTC_HCTOSYS_DEVICE.
+#
+# This service relies on 50-udev-default.rules to create symlink from hctosys specified 
+# RTC /dev/rtcx to /dev/rtc and on udev rule 90-rtc-sys-time-init.rules to create 
+# device unit dev-rtc.device when the hctosys RTC device is ready.
+[Unit]
+Description=Correct system time and sync to RTC time
+SourcePath=/proc/cmdline
+DefaultDependencies=no
+After=dev-rtc.device
+Requires=dev-rtc.device
+Before=systemd-timesyncd.service
+Conflicts=shutdown.target
```

Comment:
Missing `Before=shutdown.target`. Otherwise this does not do much.

Reply:
Thanks, added.